cypress/integration/channel/add_users_to_channel_spec.js

```diff
@@ -1,101 +1,136 @@
+
+const channelDisplayNameTest1 = 'Channel test 1';
+const channelDisplayNameTest2 = 'Channel test 2';
+let channelTest1Id;
+let channelTest2Id;
+
+function deleteCurrentChannel() {
+    cy.getCurrentChannelId().then((channelId) => {
+        cy.apiDeleteChannel(channelId);
+    });
+}
+
+function verifyMentionedUserAndProfilePopover(postId) {
+    cy.get(`#post_${postId}`).find('.mention-link').each(($el) => {
+        // # Click each username link
+        cy.wrap($el).click();
+
+        // * The profile pop-over of each user should be displayed when their username is clicked, and the pop-over is fully visible
+        const userName = $el[0].innerHTML;
+        cy.get('#user-profile-popover').should('be.visible');
+
+        // * The pop-over title  at the top of the pop-over should be the same as the username link for each user
+        const escapedUsername = userName.replace('@', '').replace('.', '-');
+        cy.get(`#userPopoverUserName-${escapedUsername}`).should('contain', userName);
+    });
+}
 
 describe('Add user to a channel', () => {
     it('CS15445 Join/leave messages (including combined): Usernames are links, open profile popovers', () => {
-        // # Login as 'user-1' and go to /
+        // # Login as 'user-1', go to / and create 'Channel test 1'
         cy.apiLogin('user-1');
         cy.visit('/');
+        cy.getCurrentTeamId().then((teamId) => {
+            cy.apiCreateChannel(teamId, 'channel-test-1', channelDisplayNameTest1).then((res) => {
+                channelTest1Id = res.body.name;
+
+                // # Select 'Channel test 1' channel on the left hand side
+                cy.get(`#sidebarItem_${channelTest1Id}`).click();
 
-        // # Select 'doloremque' channel on the left hand side
-        cy.get('#sidebarItem_saepe-5').click();
+                // * 'Channel test 1' shoul be visible at the top of the center pane
+                cy.get('#channelHeaderTitle').should('contain', 'Channel test 1');
 
-        // * 'doloremque' shoul be visible at the top of the center pane
-        cy.get('#channelHeaderTitle').should('contain', 'doloremque');
+                // # Then click it to access the drop-down menu
+                cy.get('#channelHeaderTitle').click();
 
-        // # Then click it to access the drop-down menu
-        cy.get('#channelHeaderTitle').click();
+                // * The dropdown menu of the channel header should be visible;
+                cy.get('#channelHeaderDropdownMenu').should('be.visible');
 
-        // * The dropdown menu of te channel header should be visible;
-        cy.get('#channelHeaderDropdownMenu').should('be.visible');
+                // # Click 'Add Members'
+                cy.get('#channelAddMembers').click();
 
-        // # Click 'Add Members'
-        cy.get('#channelAddMembers').click();
+                // * Assert that modal appears
+                cy.get('#addUsersToChannelModal').should('be.visible');
+                cy.wait(TIMEOUTS.TINY);
```

Comment:
@VolatianaYuliana Whats the reasoning for this wait? Modal should be visible at this point, was `#multiSelectList` not visible? If so, tacking on an assertion to `cy.get('#multiSelectList')` should suffice.

Reply:
You're right. I'm gonna get rid of it @thekiiingbob